src/box/tuple.c

```diff
@@ -177,122 +210,80 @@ tuple_next(struct tuple_iterator *it)
 	return NULL;
 }
 
-/** {{{ Bigref - allow tuple reference counter to be > 2^16 */
-
-enum {
-	BIGREF_FACTOR = 2,
-	BIGREF_MAX = UINT32_MAX,
-	BIGREF_MIN_CAPACITY = 16,
-	/**
-	 * Only 15 bits are available for bigref list index in
-	 * struct tuple.
-	 */
-	BIGREF_MAX_CAPACITY = UINT16_MAX >> 1
-};
-
 /**
- * Container for big reference counters. Contains array of big
- * reference counters, size of this array and number of non-zero
- * big reference counters. When reference counter of tuple becomes
- * more than 32767, field refs of this tuple becomes index of big
- * reference counter in big reference counter array and field
- * is_bigref is set true. The moment big reference becomes equal
- * 32767 it is set to 0, refs of the tuple becomes 32767 and
- * is_bigref becomes false. Big reference counter can be equal to
- * 0 or be more than 32767.
+ * Reference count that is uploaded or acquires to/from external storage
+ * in one iteration.
  */
-static struct bigref_list {
-	/** Free-list of big reference counters. */
-	uint32_t *refs;
-	/** Capacity of the array. */
-	uint16_t capacity;
-	/** Index of first free element. */
-	uint16_t vacant_index;
-} bigref_list;
+enum { TUPLE_UPLOAD_REFS = TUPLE_REF_MAX / 2 + 1 };
```

Comment:
Please explain in the comment why it is chosen this way.

Reply:
done